File: kennelmart/src/main/java/com/kennel/mart/kennelmart/service/AuthService.java
package com.kennel.mart.kennelmart.service;

import com.kennel.mart.kennelmart.dto.AuthResponse;
import com.kennel.mart.kennelmart.dto.LoginRequest;
import com.kennel.mart.kennelmart.dto.RegisterRequest;

/**
 * Service interface for authentication operations.
 * 
 * Defines contract for authentication use cases:
 * - User registration
 * - User login
 * - User info retrieval
 * 
 * Demonstrates OOP principle: Abstraction via interface
 */
public interface AuthService {

    /**
     * Register a new user with email and password.
     * 
     * Business Rules:
     * - Email must be unique
     * - Password must match confirmPassword
     * - Password must be at least 8 characters
     * - NU email required
     * 
     * @param request user registration data
     * @return authentication response with JWT token
     * @throws IllegalArgumentException if validation fails
     */
    AuthResponse register(RegisterRequest request);

    /**
     * Authenticate user with email and password.
     * 
     * @param request user login credentials
     * @return authentication response with JWT token
     * @throws IllegalArgumentException if credentials are invalid
     */
    AuthResponse login(LoginRequest request);
}

File: kennelmart-ui/src/services/authService.ts
import api from '../api/axios';
import { LoginRequest, RegisterRequest, AuthResponse, User } from '../types/auth';
import type { UpdateProfileRequest, ChangePasswordRequest } from '../types/auth';

export const authService = {
  async login(credentials: LoginRequest): Promise<AuthResponse> {
    const response = await api.post<AuthResponse>('/api/auth/login', credentials);
    return response.data;
  },
  async register(data: RegisterRequest): Promise<AuthResponse> {
    const response = await api.post<AuthResponse>('/api/auth/register', data);
    return response.data;
  },
  async getCurrentUser(): Promise<User> {
    const response = await api.get<User>('/api/auth/me');
    return response.data;
  },
  async updateProfile(data: UpdateProfileRequest): Promise<User> {
    const response = await api.put<User>('/api/users/me', data);
    return response.data;
  },
  async changePassword(data: ChangePasswordRequest): Promise<void> {
    await api.put('/api/users/me/password', data);
  },
};